velox/exec/tests/ArrowStreamTest.cpp

```diff
@@ -0,0 +1,240 @@
+      return privateData()->reader_;
+    }
+
+    int toError(int code) {
+      switch (code) {
+        case 0:
+          privateData()->lastError_.clear();
+          break;
+        case 1:
+          privateData()->lastError_ = "Getting next failed.";
+          break;
+        case 2:
+          privateData()->lastError_ = "Getting schema failed.";
+          break;
+        default:
+          privateData()->lastError_ = "Failed.";
+      }
+      return code;
+    }
+
+    std::shared_ptr<memory::MemoryPool> pool_;
+    struct ArrowArrayStream* stream_;
+  };
+
+  void exportArrowStream(
+      std::shared_ptr<ArrowReader> reader,
+      struct ArrowArrayStream* out) {
+    out->get_schema = ExportedArrayStream::staticGetSchema;
+    out->get_next = ExportedArrayStream::staticGetNext;
+    out->get_last_error = ExportedArrayStream::staticGetLastError;
+    out->release = ExportedArrayStream::staticRelease;
+    out->private_data = new ExportedArrayStream::PrivateData{std::move(reader)};
+  }
+};
+
+TEST_F(ArrowStreamTest, basic) {
+  vector_size_t size = 1'000;
+  std::vector<RowVectorPtr> vectors;
+  for (int32_t i = 0; i < 3; ++i) {
+    vectors.push_back(makeRowVector(
+        {makeFlatVector<bool>(
+             size, [](auto row) { return (row / 10) % 2 == 0; }, nullEvery(3)),
+         makeFlatVector<int8_t>(
+             size, [](auto row) { return row % 100; }, nullEvery(5)),
+         makeFlatVector<int16_t>(
+             size, [&](auto row) { return row * i; }, nullEvery(2)),
+         makeFlatVector<int32_t>(
+             size, [](auto row) { return row; }, nullEvery(5)),
+         makeFlatVector<int64_t>(
+             size,
+             [&](vector_size_t row) { return size * i + row; },
+             nullEvery(5)),
+         makeFlatVector<float>(
+             size, [](vector_size_t row) { return row * 1.1; }, nullEvery(5)),
+         makeFlatVector<double>(
+             size, [](vector_size_t row) { return row * 1.3; }, nullEvery(11)),
+         makeFlatVector<StringView>(
+             size,
+             [](vector_size_t row) {
+               uint64_t val = 100000000000 + (uint64_t)(row % 100);
+               return StringView(std::to_string(val));
+             },
+             nullEvery(7))}));
+  }
+  createDuckDbTable(vectors);
+  auto type = asRowType(vectors[0]->type());
+  auto reader = std::make_shared<ArrowReader>(pool_, vectors, type);
+  struct ArrowArrayStream arrowStream;
+  exportArrowStream(reader, &arrowStream);
+  auto plan = std::make_shared<core::ArrowStreamNode>(
+      "0", type, std::make_shared<ArrowArrayStream>(arrowStream));
+  assertQuery(plan, "SELECT * FROM tmp");
+}
+
+TEST_F(ArrowStreamTest, error) {
+  vector_size_t size = 1'000;
+  std::vector<RowVectorPtr> vectors;
+  for (int32_t i = 0; i < 3; ++i) {
+    vectors.push_back(makeRowVector({makeFlatVector<int32_t>(
+        size, [](auto row) { return row; }, nullEvery(5))}));
+  }
+  createDuckDbTable(vectors);
```

Comment:
This is not needed since we won't be comparing the results.

Reply:
Removed.